lib/flok.js

```diff
@@ -303,6 +303,16 @@ Flok.prototype.sortMigrationsUp = function sortMigrationsUp(callback) {
       this.migrations.splice(pos, 0, this.migrations.splice(i, 1)[0]);
     }
   }
+
+  const mBefore = filter(this.migrations, 'before');
```

Comment:
The only drawback with resorting the migrations after dependencies are checked is that something could execute before it's dependencies.  Would a safer, and equally as simple, approach be to add the migration as a dep to the old migration referenced by the `before`?

* loop over migrations with a `before` property
* add the migration's id as a dependency of the migration referenced by `before`
* sort as normal

Reply:
@seangarner is this what you mean: https://github.com/dynamicaction/flok/pull/6/commits/74c9a16e229321fab1ab3f4a84a8643fd73348c3?

Reply:
Yep.  Perfect.